server/pkg/project/project.go

```diff
@@ -157,6 +162,14 @@ func (p *Project) Unstar(uId accountdomain.UserID) {
 	}
 }
 
+func (p *Project) SetTopics(topics []string) {
+	if p == nil {
+		return
+	}
+
+	p.topics = slices.Clone(topics)
+}
+
```

Comment:
Topics should be validated, ie no duplication, no empty values ...

Reply:
Topics wont be case-sensitive, we had a discussion with Red about it and decided to not handle case-sensitivity. Thanks for considering it. 

I added e2e tests to handle duplicates and empty string values so it should cover the scenarios. Also reduces number of tests in the repo.